bundles/init/src/main/java/com/adobe/cq/commerce/demandware/init/InitServlet.java

```diff
@@ -59,26 +60,28 @@
 @SlingServlet(resourceTypes = "commerce/demandware/components/init", extensions = "html", methods = "POST", label = "Demandware Init Servlet")
 public class InitServlet extends SlingAllMethodsServlet {
 
-    private static final Logger LOG = LoggerFactory.getLogger(InitServlet.class);
+	private static final Logger LOG = LoggerFactory.getLogger(InitServlet.class);
+	private static final String PROPERTY_DW_INSTANCE_ID = "dwInstanceId";
 
-    @Reference
-    DemandwareClient demandwareClient;
+	@Reference
+    DemandwareClientProvider clientProvider;
 
     @Reference
     private Replicator replicator;
 
     @Override
-    protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response) throws
-        ServletException, IOException {
+    protected void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException {
         LOG.debug("Init Demandware Sandbox");
         ValueMap config = request.getResource().getValueMap();
-
-        // get assets and push to webdav
+	    final String instanceId = config.get(PROPERTY_DW_INSTANCE_ID, String.class);
+	    final Optional<DemandwareClient> dwClient = clientProvider.getClientForSpecificInstance(instanceId);
```

Comment:
"Optional" is not useful here - especially as we test that you have one later on..